src/server/saved_objects/client/saved_objects_client.js

```diff
@@ -175,26 +178,62 @@ export class SavedObjectsClient {
       return { saved_objects: [] };
     }
 
-    const docs = objects.map(doc => {
-      return { _type: get(doc, 'type'), _id: get(doc, 'id') };
-    });
+    const docs = objects.reduce((acc, { type, id }) => {
+      acc.push({}, createIdQuery({ type, id }));
+      return acc;
+    }, []);
+
 
-    const response = await this._withKibanaIndex('mget', { body: { docs } })
-      .then(resp => get(resp, 'docs', []).filter(resp => resp.found));
+    const response = await this._withKibanaIndex('msearch', { body: docs })
+    .then(resp => {
+      let results = [];
+      const responses = get(resp, 'responses', []);
+      responses.forEach(r => {
```

Comment:
This can be done with a single `map` rather than closing over and mutating a `results` accumulator inside a `forEach`.